src/search.cpp

```diff
@@ -189,6 +189,13 @@ Value Worker::search(Position& pos, Stack* ss, Value alpha, Value beta, Depth de
     bool  is_in_check = pos.is_in_check();
     Value static_eval = is_in_check ? -VALUE_INF : evaluate(pos);
 
+    // Internal Iterative Reductions
+    if (PV_NODE && depth >= 8 && tt_data->move == Move::none()) {        
```

Comment:
```suggestion
    if (PV_NODE && depth >= 8 && (!tt_data || tt_data->move == Move::none())) {
```

Accessing a `std::optional` that is empty is undefined behavior.